package/MDAnalysis/core/groups.py

```diff
@@ -1253,6 +1256,160 @@ def wrap(self, compound="atoms", center="com", box=None):
+        group's compounds aren't split accross periodic boundaries.
+
+        This function is most useful when atoms have been packed into the
+        primary unit cell, causing breaks mid-molecule, with the molecule then
+        appearing on either side of the unit cell. This is problematic for
+        operations such as calculating the center of mass of the molecule. ::
+
+           +-----------+       +-----------+
+           |           |       |           |
+           | 6       3 |       |         3 | 6
+           | !       ! |       |         ! | !
+           |-5-8   1-2-|  ==>  |       1-2-|-5-8
+           | !       ! |       |         ! | !
+           | 7       4 |       |         4 | 7
+           |           |       |           |
+           +-----------+       +-----------+
+
+        Parameters
+        ----------
+        compound : {'group', 'segments', 'residues', 'molecules', \
+                    'fragments'}, optional
+            Which type of component to unwrap. Note that, in any case, all
+            atoms within each compound must be interconnected by bonds, i.e.,
+            compounds must correspond to (parts of) molecules.
+        reference : {'com', 'cog', None}, optional
+            If ``'com'`` (center of mass) or ``'cog'`` (center of geometry), the
+            unwrapped compounds will be shifted so that their individual
+            reference point lies within the primary unit cell.
+            If ``None``, no such shift is performed.
+        inplace : bool, optional
+            If ``True``, coordinates are modified in place.
+
+        Returns
+        -------
+        coords : numpy.ndarray
+            Unwrapped atom coordinate array of shape ``(n, 3)``.
+
+        Raises
+        ------
+        NoDataError
+            If `compound` is ``'molecules'`` but the underlying topology does
+            not contain molecule information, or if `reference` is ``'com'``
+            but the topology does not contain masses.
+        ValueError
+            If `reference` is not one of ``'com'``, ``'cog'``, or ``None``, or
+            if `reference` is ``'com'`` and the total mass of any `compound` is
+            zero.
+
+        Note
+        ----
+        Be aware of the fact that only atoms *belonging to the group* will
+        be unwrapped! If you want entire molecules to be unwrapped, make sure
+        that all atoms of these molecules are part of the group.\n
+        An AtomGroup containing all atoms of all fragments in the group ``ag``
+        can be created with::
+
+          all_frag_atoms = sum(ag.fragments)
+
+
+        See Also
+        --------
+        :func:`~MDAnalysis.lib.mdamath.make_whole`,
+        :meth:`~MDAnalysis.AtomGroup.wrap`,
+        :meth:`~MDAnalysis.AtomGroup.pack_into_box`,
+        :func:`~MDanalysis.lib.distances.apply_PBC`
+
+
+        .. versionadded:: 0.20.0
+        """
+        atoms = self.atoms
+        # bail out early if no bonds in topology:
```

Comment:
there's a `requires` decorator that can do this type of thing (and create uniform error messages)

Reply:
@richardjgowers I know. The `@requires()` decorator only works on AtomGroups, that's why it's not used here.

Reply:
One could extend the `@requires` decorator to check for `GroupBase` instances instead of `AtomGroup` and then check the group's `.atoms` member for the given attribute(s). However, I don't like this solution since it
1. doesn't allow to check for `ResiduGroup` or `SegmentGroup` attributes and
2. more importantly, isn't lightweight anymore since it involves object instantiation when accessing the `.atoms` member.

The check I implemented doesn't instantiate additional objects and its impact on performance should therefore be negligible.

Reply:
Hmm, really we could have something like `assert 'bonds' in u.topology` which works... I'll raise an issue.  I just like error messages being uniform, so hitting eg 'no masses' always looks similar